python/src/nanoarrow/_lib.pyx

```diff
@@ -1177,11 +1429,395 @@ cdef class CBufferView:
+
+        self._refresh_view_if_needed()
+        return f"CBuffer({_lib_utils.buffer_view_repr(self._view)})"
+
+
+cdef class CBufferBuilder(CBuffer):
+    """Wrapper around writable owned buffer CPU content"""
+
+    def reserve_bytes(self, int64_t additional_bytes):
+        self._assert_valid()
+        self._assert_buffer_count_zero()
+        cdef int code = ArrowBufferReserve(self._ptr, additional_bytes)
+        Error.raise_error_not_ok("ArrowBufferReserve()", code)
+        return self
+
+    def write(self, content):
+        self._assert_valid()
+        self._assert_buffer_count_zero()
+
+        cdef Py_buffer buffer
+        cdef int64_t out
+        PyObject_GetBuffer(content, &buffer, PyBUF_ANY_CONTIGUOUS)
+
+        cdef int code = ArrowBufferReserve(self._ptr, buffer.len)
+        if code != NANOARROW_OK:
+            PyBuffer_Release(&buffer)
+            Error.raise_error("ArrowBufferReserve()", code)
+
+        code = PyBuffer_ToContiguous(
+            self._ptr.data + self._ptr.size_bytes,
+            &buffer,
+            buffer.len,
+            # 'C' (not sure how to pass a character literal here)
+            43
+        )
+        out = buffer.len
+        PyBuffer_Release(&buffer)
+        Error.raise_error_not_ok("PyBuffer_ToContiguous()", code)
+
+        self._ptr.size_bytes += out
+        return out
+
+    def write_values(self, obj):
+        self._assert_valid()
+
+        if self._data_type == NANOARROW_TYPE_BOOL:
+            return self._write_bits(obj)
+
+        cdef int64_t n_values = 0
+        struct_obj = Struct(self._format)
+        pack = struct_obj.pack
+        write = self.write
+
+        if self._data_type in (NANOARROW_TYPE_INTERVAL_DAY_TIME,
+                               NANOARROW_TYPE_INTERVAL_MONTH_DAY_NANO):
+            for item in obj:
+                n_values += 1
+                write(pack(*item))
+        else:
+            for item in obj:
+                n_values += 1
+                write(pack(item))
+
+        return n_values
+
+    cdef _write_bits(self, obj):
+        if self._ptr.size_bytes != 0:
+            raise NotImplementedError("Append to bitmap that has already been appended to")
+
+        cdef char buffer_item = 0
+        cdef int buffer_item_i = 0
+        cdef int code
+        cdef int64_t n_values = 0
+        for item in obj:
+            n_values += 1
+            if item:
+                buffer_item |= (<char>1 << buffer_item_i)
+
+            buffer_item_i += 1
+            if buffer_item_i == 8:
+                code = ArrowBufferAppendInt8(self._ptr, buffer_item)
+                Error.raise_error_not_ok("ArrowBufferAppendInt8()", code)
+                buffer_item = 0
+                buffer_item_i = 0
+
+        if buffer_item_i != 0:
+            code = ArrowBufferAppendInt8(self._ptr, buffer_item)
+            Error.raise_error_not_ok("ArrowBufferAppendInt8()", code)
+
+        return n_values
+
+    def finish(self):
+        return self
```

Comment:
This means that it returns a BufferBuilder, which has some additional methods to further append to the Buffer?